towerlib/towerlib.py

```diff
@@ -1997,24 +2011,579 @@ def schedules(self):
 
     def get_schedule_by_id(self, id_):
         """Retrieves a schedule by id.
-
         Args:
             id_: The id of the schedule to retrieve.
-
         Returns:
             Schedule: The schedule if a match is found else None.
-
         """
         return next(self.schedules.filter({'id': id_}), None)
 
     def get_schedule_by_name(self, name):
         """Retrieves an schedule by name.
-
         Args:
             name: The name of the schedule to retrieve.
-
         Returns:
             Schedule: The schedule if a match is found else None.
-
         """
         return next(self.schedules.filter({'name__iexact': name}), None)
+
+    #    ****** High level methods for Spirit/21 ******    #
+
+    def update_all_projects(self):
+        """Update all the projects in ansible tower one by one.
+        """
+        for project in self.projects:
+            project.update()
+
+    def update_project_by_id(self, project_id):
+        """Update the ansible tower project with given project id.
+
+        Args:
+            project_id: The id of the project, which is to be updated.
+
+        Returns:
+            list: List of response of api request as json on success, False otherwise.
+
+        """
+        project = self.get_project_by_id(project_id)
+        return project.update()
+
+    def update_project_by_name(self, organization, project_name):
+        """Update the ansible tower project with given project name.
+
+        Args:
+            organization: The name of the organization.
+            project_name: The name of the project, which is to be updated.
+
+        Returns:
+            dict: dict of response of api request as json on success, None otherwise.
+
+        """
+
+        project = self.get_organization_project_by_name(organization, project_name)
+        updated_project = self.update_project_by_id(project.id)
+        return updated_project
+
+    def update_project_by_scm_url(self, scm_url):
```

Comment:
This can affect all the projects under the whole service which effectively breaks down the boundaries set by organizations and it makes me uncomfortable. Please lower the scope of this by making it work on the scope of an organization.

Reply:
Sometimes, we need to send SCM update request to all the projects, which have the same url.

Reply:
Radius is reduced to an org.